File: day01/part2.lua
local calories = {}
local current = 0

for line in io.lines() do
  if line == "" then
    table.insert(calories, current)
    current = 0
  else
    current = current + tonumber(line)
  end
end

function reverse(n1, n2)
  return n1 > n2
end

table.sort(calories, reverse)
print(calories[1] + calories[2] + calories[3])

File: day01/part2.rs
use std::io;
use std::io::BufRead;

fn main() -> io::Result<()> {
    let mut current = 0;

    let mut v: Vec::<i32> = Vec::new();

    let stdin = io::stdin();
    {
        let handle = stdin.lock();
        for line in handle.lines() {
            if let Ok(calorie) = line {
                if calorie == "" {
                    v.push(current);
                    current = 0;
                } else {
                    let c: i32 = calorie.parse().unwrap();
                    current = current + c;
                }
            }
        }
    }

    v.sort_by(|a, b| b.cmp(a));
    println!("{}", v[0] + v[1] + v[2]);
    Ok(())
}
